pypeit/calibrations.py

```diff
@@ -1190,16 +1260,65 @@ def get_tilts(self):
         self.wavetilts.to_file()
         return self.wavetilts
 
-    def run_the_steps(self):
+    def process_load_selection(self, frame, cal_file, force):
+        """
+        Process how pypeit should use any pre-existing calibration files.
+
+        If loading is requested but the calibration file (``cal_file``) does
+        not exist, ``self.success`` is set to False, and None is returned.
+
+        Args:
+            frame (:obj:`dict`):
+                A dictionary with two elements: ``type`` is the string
+                defining the frame type and ``class`` is the pypeit class
+                used to load the pre-existing calibration file.
+            cal_file (:obj:`str`, `Path`_):
+                Path to the calibration file.
+            force (:obj:`str`):
+                Defines how to treat a pre-existing calibration file.  Must
+                be one of the following options:
+                    
+                    - ``'remake'``: Force the calibration be remade.
+
+                    - ``'reload'``: Reload the frame if it exists.
+
+                    - ``None``: Load the existing frame if it exists and
+                    ``self.reuse_calibs=True``.
+
+        Returns:
+            :obj:`object`:  Either the loaded calibration object or None.
+        """
+        if force not in [None, 'remake', 'reload']:
+            msgs.error(f'`force` keyword must be None, remake, or reload, not {force}')
+        if force == 'remake':
+            return None
+        _cal_file = Path(cal_file).absolute()
+        if force == 'reload' and not _cal_file.exists():
+            msgs.warn(f"{_cal_file} does not exist; cannot reload "
+                    f"{frame['class'].__name__} calibration.")
+            self.success = False
+            return None
+        if force == 'reload' or (self.reuse_calibs and _cal_file.exists()): 
+            return frame['class'].from_file(_cal_file, chk_version=self.chk_version)
+
+    def run_the_steps(self, stop_at_step:str=None):
         """
         Run full the full recipe of calibration steps.
         """
         self.success = True
         for step in self.steps:
-            getattr(self, f'get_{step}')()
+            #self.run_one_step(step)
+            if stop_at_step is not None and step == stop_at_step:
+                force = 'remake'
+            else:
+                force = None
+            getattr(self, f'get_{step}')(force=force)
             if not self.success:
                 self.failed_step = f'get_{step}'
                 return
+            if stop_at_step is not None and step == stop_at_step:
+                msgs.info(f"Calibrations stopping at {stop_at_step}") 
```

Comment:
Can you move this `msgs.info` to the `if stop_at_step is not None and step == stop_at_step:` above, so that this `if statement` is not repeated twice?

Reply:
well, it's meant to inform the user we are stopping and 
it should only print when successful.  

But I've added another comment above.